src/Evolution/Systems/GeneralizedHarmonic/Equations.hpp

```diff
@@ -127,4 +142,150 @@ struct ComputeNormalDotFluxes {
+ * that are positive ("outgoing", i.e. leaving the element); and
+ * \f$\Lambda^{\rm neg}\f$ is a diagonal matrix whose nonzero entries are
+ * the average characteristic speeds that are negative ("incoming", i.e.
+ * entering the element). If a characteristic field
+ * \f$U^{\rm int}\f$ has a characteristic speed \f$v^{\rm int}\f$ and the same
+ * field in the exterior \f$U^{\rm ext}\f$ has speed \f$v^{\rm ext}\f$, then
+ * the average characteristic speed is
+ * \f$v^{\rm avg} = (1/2)(v^{\rm int} + v^{\rm ext})\f$.
+ *
+ * This function implements the upwind flux for the generalized harmonic
+ * system. First, it computes the characteristic variables using i) the
+ * evolved variables from the interior and ii) the evolved variables from the
+ * exterior. Then, it computes \f$\Lambda^{\rm pos}\f$ and
+ * \f$\Lambda^{\rm neg}\f$ from the average characteristic speeds.
+ * Then, it computes the combination
+ * \f$\Lambda^{\rm pos} S^{-1} u^{\rm int}
+ * + \Lambda^{\rm neg} S^{-1} u^{\rm ext}\f$. Finally, it
+ * applies \f$S\f$ by converting the result back from characteristic to
+ * evolved variables, using the unit normal vector of the element and
+ * the average value of the field
+ * \f$\gamma_2^{\rm avg} = (1/2)(\gamma_2^{\rm int} + \gamma_2^{\rm ext})\f$,
+ * where here \f$\gamma_2^{\rm int}\f$ is the value of \f$\gamma_2\f$
+ * in the interior and \f$\gamma_2^{\rm ext}\f$ is the value of
+ * \f$\gamma_2\f$ in the exterior.
+ *
+ * \warning as currently implemented, this flux does not satisfy the
+ * generalized Rankine-Hugoniot condition
+ * \f$D^{\rm int}([S^{-1} u]^{\rm int}, [S^{-1} u]^{\rm ext})
+ * = - D^{\rm ext}([S^{-1} u]^{\rm ext}, [S^{-1} u]^{\rm int})\f$,
+ * which enforces that the flux leaving the interior is the flux entering
+ * the exterior and vice versa. This condition is necessary for a
+ * well-balanced scheme. Use this flux with caution; so far it has
+ * been tested for evolving aligned-spin Kerr-Schild in generalized harmonic.
+ */
+template <size_t Dim>
+struct UpwindFlux {
+ public:
+  using options = tmpl::list<>;
+  static constexpr OptionString help = {
+      "Computes the generalized harmonic upwind flux."};
+
+  // clang-tidy: non-const reference
+  void pup(PUP::er& /*p*/) noexcept {}  // NOLINT
+
+  // This is the data needed to compute the numerical flux.
+  // `dg::SendBoundaryFluxes` calls `package_data` to store these tags in a
+  // Variables. Local and remote values of this data are then combined in the
+  // `()` operator.
+  using package_tags = tmpl::list<
+      gr::Tags::SpacetimeMetric<Dim, Frame::Inertial, DataVector>,
+      Tags::Pi<Dim, Frame::Inertial>, Tags::Phi<Dim, Frame::Inertial>,
+      gr::Tags::Lapse<DataVector>,
+      gr::Tags::Shift<Dim, Frame::Inertial, DataVector>,
+      gr::Tags::InverseSpatialMetric<Dim, Frame::Inertial, DataVector>,
+      Tags::ConstraintGamma1, Tags::ConstraintGamma2,
+      ::Tags::Normalized<::Tags::UnnormalizedFaceNormal<Dim, Frame::Inertial>>>;
```

Comment:
In unrelated code I came across an issue that could affect your code here. There is code in `FluxCommunicationTypes.hpp` that tries to remove prefixes from these `package_tags`, which fails for your new `Tags::Normalized` (it's actually the `Variables` copy constructor that does that). Before merging this PR, please check if the code works with your GH executable. If you run into the same issue, I suggest to do what I have suggested before: make a new `FaceNormal` tag that is private to the class and use it to package the unit normal.

Reply:
Thanks for the heads up! I just rebased on the latest develop, and this PR works with the GH executable, so I think we don't need to rework this before merging